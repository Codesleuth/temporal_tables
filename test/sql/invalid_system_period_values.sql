SET client_min_messages TO error;

CREATE TABLE invalid_system_period_values (a bigint, "b b" date, sys_period tstzrange);

INSERT INTO invalid_system_period_values (a, sys_period) VALUES (1, NULL);
INSERT INTO invalid_system_period_values (a, sys_period) VALUES (2, 'empty');
INSERT INTO invalid_system_period_values (a, sys_period) VALUES (3, tstzrange(NULL, CURRENT_TIMESTAMP));

CREATE TABLE invalid_system_period_values_history (a bigint, "b b" timestamp, sys_period tstzrange);

CREATE TRIGGER versioning_trigger
BEFORE INSERT OR UPDATE OR DELETE ON invalid_system_period_values
FOR EACH ROW EXECUTE PROCEDURE versioning('sys_period', 'invalid_system_period_values_history', true);

DELETE FROM invalid_system_period_values WHERE a = 1;
DELETE FROM invalid_system_period_values WHERE a = 2;
DELETE FROM invalid_system_period_values WHERE a = 3;