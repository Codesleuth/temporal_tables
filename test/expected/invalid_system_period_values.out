SET client_min_messages TO error;
CREATE TABLE invalid_system_period_values (a bigint, "b b" date, sys_period tstzrange);
INSERT INTO invalid_system_period_values (a, sys_period) VALUES (1, NULL);
INSERT INTO invalid_system_period_values (a, sys_period) VALUES (2, 'empty');
INSERT INTO invalid_system_period_values (a, sys_period) VALUES (3, tstzrange(NULL, CURRENT_TIMESTAMP));
CREATE TABLE invalid_system_period_values_history (a bigint, "b b" timestamp, sys_period tstzrange);
CREATE TRIGGER versioning_trigger
BEFORE INSERT OR UPDATE OR DELETE ON invalid_system_period_values
FOR EACH ROW EXECUTE PROCEDURE versioning('sys_period', 'invalid_system_period_values_history', true);
DELETE FROM invalid_system_period_values WHERE a = 1;
ERROR:  system period column "sys_period" of relation "invalid_system_period_values" must not be null
DELETE FROM invalid_system_period_values WHERE a = 2;
ERROR:  system period column "sys_period" of relation "invalid_system_period_values" contains invalid value
DETAIL:  valid ranges must be non-empty and unbounded on the high side
DELETE FROM invalid_system_period_values WHERE a = 3;
ERROR:  system period column "sys_period" of relation "invalid_system_period_values" contains invalid value
DETAIL:  valid ranges must be non-empty and unbounded on the high side
